control/subscription_group.go

```diff
@@ -386,72 +389,53 @@ func (s *subscriptionGroup) process(id string) (serrs []serror.SnapError) {
 
 func (s *subscriptionGroup) subscribePlugins(id string,
 	plugins []core.SubscribedPlugin) (serrs []serror.SnapError) {
-	for _, sub := range plugins {
+	plgs := make([]*loadedPlugin, len(plugins))
+	// First range through plugins to verify if all required plugins
+	// are available
+	for i, sub := range plugins {
+		plg, err := s.pluginManager.get(key(sub))
+		if err != nil {
+			fmt.Println(err)
+			serrs = append(serrs, serror.New(err))
+			return
+		}
+		plgs[i] = plg
+	}
+
+	// If all plugins are available, subscribe to pools and start
+	// plugins as needed
+	for _, plg := range plgs {
 		controlLogger.WithFields(log.Fields{
-			"name":    sub.Name(),
-			"type":    sub.TypeName(),
-			"version": sub.Version(),
+			"name":    plg.Name(),
+			"type":    plg.TypeName(),
+			"version": plg.Version(),
 			"_block":  "subscriptionGroup.subscribePlugins",
 		}).Debug("plugin subscription")
-		if sub.Version() < 1 {
```

Comment:
Are we losing functionality in the case where the veresion < 1 or was this not needed?